packages/style-engine/class-wp-style-engine-css-declarations.php

```diff
@@ -127,15 +127,31 @@ public function get_declarations_string( $should_prettify = false, $indent_count
 		$indent              = $should_prettify ? str_repeat( "\t", $indent_count ) : '';
 		$suffix              = $should_prettify ? ' ' : '';
 		$suffix              = $should_prettify && $indent_count > 0 ? "\n" : $suffix;
+		$spacer              = $should_prettify ? ' ' : '';
 
 		foreach ( $declarations_array as $property => $value ) {
-			$spacer               = $should_prettify ? ' ' : '';
-			$filtered_declaration = esc_html( safecss_filter_attr( "{$property}:{$spacer}{$value}" ) );
+			if (
+				0 === strpos( $property, '--' ) || // Account for CSS variables.
+				( 'display' === $property && 'none' !== $value ) || // Account for "display" when the value is not "none".
+				0 === strpos( $value, 'min(' ) || // Account for min() values.
+				0 === strpos( $value, 'max(' ) || // Account for max() values.
+				0 === strpos( $value, 'clamp(' ) // Account for clamp() values.
+			) {
+				$declarations_output .= "{$indent}{$property}:{$spacer}{$value};$suffix";
+				continue;
+			}
+			$filtered_declaration = safecss_filter_attr( "{$property}:{$spacer}{$value};" );
```

Comment:
Maybe we can use [`safecss_filter_attr_allow_css`](https://developer.wordpress.org/reference/functions/safecss_filter_attr/) filter and regex from [55966](https://core.trac.wordpress.org/ticket/55966). Then we can remove the compat function after the core patch is merged.

Reply:
Oh, that's a good point. Ideally we'd be deferring as much as possible to `safecss_filter_attr` rather than adding too much extra handling in the style engine directly.

I've opened up a potential alternative PR in #43004 to play with this idea.